src/problem.jl

```diff
@@ -0,0 +1,82 @@
+"""
+    StateSpaceProblem{isinplace, dType, uType,tType,noiseType,oType, lType} <: AbstractStateSpaceProblem{isinplace}
+
+A state space problem contains all the information necessary to simulate a state space
+process.
+"""
+struct StateSpaceProblem{isinplace, dType, uType,tType,noiseType,oType, lType} <: AbstractStateSpaceProblem{isinplace}
+    f::Function # Evolution
+    g::Function # Noise function
+    h::Function # Observation function
+    D::dType # Noise distribution
+    u0::uType # Initial condition
+    tspan::tType # Timespan to use
+    noise::noiseType # The noise to use
+    observables::oType # Data to use, if any
+    likelihood::lType # Likelihood function, if any
+end
+
+"""
+    StateSpaceProblem(f, g, h, D, u0, tspan, noise, observables, likelihood=((x...) -> 0.0))
+"""
+function StateSpaceProblem(f, g, h, D, u0, tspan, noise, observables, likelihood=((x...) -> 0.0))
+    return StateSpaceProblem{
+        Val(false), # Default to out-of-place
```

Comment:
Maybe you could use `SciMLBase.isinplace` in the `StateSpaceProblem(...)` constructor (non-inferrable but convenient) and forward it to an inferrable constructor `StateSpaceProblem{iip}(...)`? Similar to https://github.com/SciML/SciMLBase.jl/blob/ccaba96f4d7e29e9980cd4cd7270086fc5e542d6/src/problems/ode_problems.jl#L71-L73.

Reply:
Perfect, love it! I'll add it in.